cmd/edenController.go

```diff
@@ -0,0 +1,463 @@
+	}
+	if modeURL, ok = params["URL"]; !ok {
+		return "", "", fmt.Errorf("cannot parse modeURL (not [file|proto|adam|zedcloud]://<URL>): %s", controllerMode)
+	}
+	return
+}
+
+var controllerCmd = &cobra.Command{
+	Use:   "controller",
+	Short: "interact with controller",
+	Long:  `Interact with controller.`,
+}
+
+var edgeNode = &cobra.Command{
+	Use:   "edge-node",
+	Short: "manage EVE instance",
+	Long:  `Manage EVE instance.`,
+}
+
+var edgeNodeReboot = &cobra.Command{
+	Use:   "reboot",
+	Short: "reboot EVE instance",
+	Long:  `reboot EVE instance.`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		assingCobraToViper(cmd)
+		if baseOSVersionFlag := cmd.Flags().Lookup("os-version"); baseOSVersionFlag != nil {
+			if err := viper.BindPFlag("eve.base-tag", baseOSVersionFlag); err != nil {
+				log.Fatal(err)
+			}
+		}
+		_, err := utils.LoadConfigFile(configFile)
+		if err != nil {
+			return fmt.Errorf("error reading configFile: %s", err.Error())
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		modeType, modeURL, err := getControllerMode()
+		if err != nil {
+			log.Fatal(err)
+		}
+		log.Infof("Mode type: %s", modeType)
+		log.Infof("Mode url: %s", modeURL)
+		var changer configChanger
+		switch modeType {
+		case "file":
+			changer = &fileChanger{fileConfig: modeURL}
+		case "adam":
+			changer = &adamChanger{adamUrl: modeURL}
+
+		default:
+			log.Fatalf("Not implemented type: %s", modeType)
+		}
+
+		ctrl, dev, err := changer.getControllerAndDev()
+		if err != nil {
+			log.Fatalf("getControllerAndDev error: %s", err)
+		}
+		rebootCounter, _ := dev.GetRebootCounter()
+		dev.SetRebootCounter(rebootCounter+1, true)
+		if err = changer.setControllerAndDev(ctrl, dev); err != nil {
+			log.Fatalf("setControllerAndDev error: %s", err)
+		}
+		log.Info("Reboot request has been sent")
+	},
+}
+
+var edgeNodeEVEImageUpdate = &cobra.Command{
+	Use:   "eveimage-update",
+	Short: "update EVE image",
+	Long:  `Update EVE image.`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		assingCobraToViper(cmd)
+		if baseOSVersionFlag := cmd.Flags().Lookup("os-version"); baseOSVersionFlag != nil {
+			if err := viper.BindPFlag("eve.base-tag", baseOSVersionFlag); err != nil {
+				log.Fatal(err)
+			}
+		}
+		viperLoaded, err := utils.LoadConfigFile(configFile)
+		if err != nil {
+			return fmt.Errorf("error reading configFile: %s", err.Error())
+		}
+		if viperLoaded {
+			eserverIP = viper.GetString("eden.eserver.ip")
+			eserverPort = viper.GetString("eden.eserver.port")
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		rootFsPath, err := utils.GetFileFollowLinks(baseOSImage)
+		if err != nil {
+			log.Fatalf("GetFileFollowLinks: %s", err)
+		}
+		modeType, modeURL, err := getControllerMode()
+		if err != nil {
+			log.Fatal(err)
+		}
+		log.Infof("Mode type: %s", modeType)
+		log.Infof("Mode url: %s", modeURL)
+		var changer configChanger
```

Comment:
Feels like these repeated blocks need to be refactored down to a utility method so we don't have code duplication between all these different viper commands.